seriate.py

```diff
@@ -21,6 +21,21 @@ class IncompleteSolutionError(Exception):
     pass
 
 
+class InvalidDistanceValues(ValueError):
+    """Indicate that the distance array contains invalid values."""
+
+
+def _validate_data(dists: numpy.ndarray):
+    """Check dists contains valid values."""
+    try:
+        if numpy.isinf(dists).any():
+            raise InvalidDistanceValues("Data contains inf values.")
+        elif numpy.isnan(dists).any():
+            raise InvalidDistanceValues("Data contains NaN values.")
+    except Exception as e:
+        raise InvalidDistanceValues("Data contains non numeric data.")
```

Comment:
Why try-catch? We raise nice fine-grained errors inside.

Reply:
`np.isnan` and `np.isinf` can fail in case the numpy array contains data that is not compatible with those functions. There can be a case where an array contains invalid values that throw an error when calling `np.isnan` or `np.isinf`, like strings or python objects.